pkg/scheduler/framework/preemption/preemption.go

```diff
@@ -442,120 +442,85 @@ func pickOneNodeForPreemption(nodesToVictims map[string]*extenderv1.Victims) str
 			// priority (and similar scenarios).
 			sumPriorities += int64(corev1helpers.PodPriority(pod)) + int64(math.MaxInt32+1)
 		}
-		if sumPriorities < minSumPriorities {
-			minSumPriorities = sumPriorities
-			lenNodes1 = 0
-		}
-		if sumPriorities == minSumPriorities {
-			minNodes1[lenNodes1] = node
-			lenNodes1++
-		}
+		// The smaller the sumPriorities, the higher the score.
+		return -sumPriorities
 	}
-	if lenNodes1 == 1 {
-		return minNodes1[0]
+	minNumPodsScoreFunc := func(node string) int64 {
+		// The smaller the length of pods, the higher the score.
+		return -int64(len(nodesToVictims[node].Pods))
 	}
-
-	// There are a few nodes with minimum highest priority victim and sum of priorities.
-	// Find one with the minimum number of pods.
-	minNumPods := math.MaxInt32
-	lenNodes2 = 0
-	for i := 0; i < lenNodes1; i++ {
-		node := minNodes1[i]
-		numPods := len(nodesToVictims[node].Pods)
-		if numPods < minNumPods {
-			minNumPods = numPods
-			lenNodes2 = 0
-		}
-		if numPods == minNumPods {
-			minNodes2[lenNodes2] = node
-			lenNodes2++
-		}
-	}
-	if lenNodes2 == 1 {
-		return minNodes2[0]
-	}
-
-	// There are a few nodes with same number of pods.
-	// Find the node that satisfies latest(earliestStartTime(all highest-priority pods on node))
-	latestStartTime := util.GetEarliestPodStartTime(nodesToVictims[minNodes2[0]])
-	if latestStartTime == nil {
-		// If the earliest start time of all pods on the 1st node is nil, just return it,
-		// which is not expected to happen.
-		klog.ErrorS(errors.New("earliestStartTime is nil for node"), "Should not reach here", "node", klog.KRef("", minNodes2[0]))
-		return minNodes2[0]
-	}
-	nodeToReturn := minNodes2[0]
-	for i := 1; i < lenNodes2; i++ {
-		node := minNodes2[i]
+	latestStartTimeScoreFunc := func(node string) int64 {
 		// Get earliest start time of all pods on the current node.
 		earliestStartTimeOnNode := util.GetEarliestPodStartTime(nodesToVictims[node])
 		if earliestStartTimeOnNode == nil {
 			klog.ErrorS(errors.New("earliestStartTime is nil for node"), "Should not reach here", "node", klog.KRef("", node))
-			continue
+			return int64(math.MinInt64)
+		}
+		// The bigger the earliestStartTimeOnNode, the higher the score.
+		return earliestStartTimeOnNode.UnixNano()
+	}
+
+	scoreFuncs := []func(string) int64{
+		minNumPDBViolatingScoreFunc,
+		// There are more than one node with minimum number PDB violating pods. Find
+		// the one with minimum highest priority victim.
+		minHighestPriorityScoreFunc,
+		// There are a few nodes with minimum highest priority victim. Find the
+		// smallest sum of priorities.
+		minSumPrioritiesScoreFunc,
+		// There are a few nodes with minimum highest priority victim and sum of priorities.
+		// Find one with the minimum number of pods.
+		minNumPodsScoreFunc,
+		// There are a few nodes with same number of pods.
+		// Find the node that satisfies latest(earliestStartTime(all highest-priority pods on node))
+		latestStartTimeScoreFunc,
```

Comment:
Revise the wording if you'd like. (I'm not a good English writer.

```suggestion
	// Each scoreFuncs gives a high score on preferred Node.
	// The functions are executed from the top,
	// and if there are still ties after execution, then we execute next scoreFunc.
	scoreFuncs := []func(string) int64{
		// A node with a minimum number of PDB is preferable.
		minNumPDBViolatingScoreFunc,
		// A node with a minimum highest priority victim is preferable.
		minHighestPriorityScoreFunc,
		// A node with the smallest sum of priorities is preferable.
		minSumPrioritiesScoreFunc,
		// A node with the minimum number of pods is preferable.
		minNumPodsScoreFunc,
		// A node with the latest start time of all highest priority victims is preferable.
		latestStartTimeScoreFunc,
		// If there are still ties, then the first Node in the list is selected.
```

Reply:
fixed